Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -8169,21 +8169,18 @@ bool SpeculativeJIT::compileStrictEq(Node* node)
         return false;
     }
 
-    if (node->isBinaryUseKind(OtherUse)
+    if (node->isBinaryUseKind(BooleanUse, UntypedUse)
+        || node->isBinaryUseKind(UntypedUse, BooleanUse)
```

Comment:
Nit: Maybe we should add a `node->isReflexiveBinaryUseKind(X, Y)` for this?

Reply:
Sounds good. Done.

Reply:
Actually the `reflexive binary relation` on the set `{x, y}` means, each element is related to itself. Given that the minimal reflexive relation on `{x, y}` is `R = {(x, x), (y, y)}`. So, I think the best reasonable name for this case is `is SymmetricBinaryUseKind` which is based on the symmetric relation.

Reply:
Yeah you're right, symmetric is what I meant